packages/arb-token-bridge-ui/src/components/common/NetworkSelectionContainer.tsx

```diff
@@ -0,0 +1,373 @@
+            {chainType} Chain, {nativeTokenData?.symbol ?? 'ETH'} is the native
+            gas token
+          </p>
+        )}
+      </div>
+    </button>
+  )
+}
+
+function AddCustomOrbitChainButton() {
+  const [, setQueryParams] = useArbQueryParams()
+  const { isTestnetMode } = useIsTestnetMode()
+
+  if (!isTestnetMode) {
+    return null
+  }
+
+  return (
+    <button
+      className="text-sm underline"
+      onClick={() => setQueryParams({ settingsOpen: true })}
+    >
+      <span>Add Custom Orbit Chain</span>
+    </button>
+  )
+}
+
+function NetworksPanel({
+  networks,
+  onNetworkRowClick,
+  close
+}: {
+  networks: ChainId[]
+  onNetworkRowClick: (value: Chain) => void
+  close: (focusableElement?: HTMLElement) => void
+}) {
+  const [errorMessage, setErrorMessage] = useState('')
+  const [networkSearched, setNetworkSearched] = useState('')
+  const [debouncedNetworkSearched, setDebouncedNetworkSearched] = useState('')
+  const listRef = useRef<List>(null)
+  const { isTestnetMode } = useIsTestnetMode()
+
+  const testnetToggleClassNames = {
+    switch:
+      'ui-checked:bg-black/20 ui-not-checked:bg-black/20 [&_span]:ui-not-checked:bg-black'
+  }
+
+  useEffect(() => {
+    listRef.current?.recomputeRowHeights()
+  }, [isTestnetMode])
+
+  useDebounce(
+    () => {
+      setDebouncedNetworkSearched(networkSearched)
+    },
+    200,
+    [networkSearched]
+  )
+
+  function getRowHeight({ index }: { index: number }) {
+    const rowItemOrChainId = networksToShowWithChainTypeInfo[index]
+    if (!rowItemOrChainId) {
+      return 0
+    }
+    if (typeof rowItemOrChainId === 'object') {
+      return 65
+    }
+    const rowItem = getBridgeUiConfigForChain(rowItemOrChainId)
+    if (rowItem.description) {
+      return 90
+    }
+    return 52
+  }
+
+  const networksToShowWithChainTypeInfo = useMemo(() => {
+    const _networkSearched = debouncedNetworkSearched.trim().toLowerCase()
+
+    if (_networkSearched) {
+      return networks.filter(network => {
+        const networkName =
+          getBridgeUiConfigForChain(network).network.name.toLowerCase()
+        return networkName.includes(_networkSearched)
+      })
+    }
+
+    const coreNetworks = networks.filter(
+      chainId => !isNetwork(chainId).isOrbitChain
+    )
+    const orbitNetworks = networks.filter(
+      chainId => isNetwork(chainId).isOrbitChain
+    )
+
+    return [
+      chainGroupInfo.core,
+      ...coreNetworks,
+      chainGroupInfo.orbit,
+      ...orbitNetworks
+    ]
+  }, [debouncedNetworkSearched, networks])
+
+  const rowRenderer = useCallback(
+    (virtualizedProps: ListRowProps) => {
+      const networkOrChainTypeInfo =
+        networksToShowWithChainTypeInfo[virtualizedProps.index]
+      if (!networkOrChainTypeInfo) {
+        return null
+      }
+
+      // Chain Type Info row
+      if (typeof networkOrChainTypeInfo === 'object') {
+        const isCoreGroup = networkOrChainTypeInfo.name === 'CORE CHAINS'
```

Comment:
if we do that then 'CORE CHAINS' and 'ORBIT CHAINS' should be an enum

Reply:
We could probably store them as `Core chains` & `Orbit chains` only only uppercase with tailwind as needed